src/FluentValidation/AbstractValidator.cs

```diff
@@ -31,16 +31,130 @@ namespace FluentValidation {
+		/// to that value.
+		/// </para>
+		/// <para>
+		/// If set to the deprecated <see cref="CascadeMode.StopOnFirstFailure"/>, then <see cref="AbstractValidator.ClassLevelCascadeMode"/>
+		/// is set to <see cref="CascadeMode.Continue"/>, and <see cref="AbstractValidator.RuleLevelCascadeMode"/>
+		/// is set to <see cref="CascadeMode.Stop"/>.
+		/// This results in the same behaviour as before this property was deprecated.
+		/// </para>
 		/// </summary>
+		[Obsolete($"Use {nameof(ClassLevelCascadeMode)} and/or {nameof(RuleLevelCascadeMode)} instead. " +
+			"CascadeMode will be removed in a future release. " +
+			"For more details, see https://docs.fluentvalidation.net/en/latest/conditions.html#setting-the-cascade-mode")]
 		public CascadeMode CascadeMode {
-			get => _cascadeMode();
-			set => _cascadeMode = () => value;
+			get {
+				if (ClassLevelCascadeMode == RuleLevelCascadeMode) {
+					return ClassLevelCascadeMode;
+				}
+				else if (ClassLevelCascadeMode == CascadeMode.Continue && RuleLevelCascadeMode == CascadeMode.Stop) {
+					return CascadeMode.StopOnFirstFailure;
+				}
+				else {
+					throw new Exception(
+						$"There is no conversion to a single {nameof(CascadeMode)} value from the current combination of " +
+							$"{nameof(ClassLevelCascadeMode)} and {nameof(RuleLevelCascadeMode)}. " +
+							$"Please use these properties instead of the deprecated {nameof(CascadeMode)} going forward.");
+				}
+			}
+			set {
+				ClassLevelCascadeMode = value == CascadeMode.StopOnFirstFailure
+					? CascadeMode.Continue
+					: value;
+
+				RuleLevelCascadeMode = value == CascadeMode.StopOnFirstFailure
+					? CascadeMode.Stop
+					: value;
+			}
+		}
+#pragma warning restore 618
+
+		/// <summary>
+		/// <para>
+		/// Sets the cascade behaviour <i>in between</i> rules in this validator.
+		/// This overrides the default value set in <see cref="ValidatorConfiguration.DefaultClassLevelCascadeMode"/>.
+		/// </para>
+		/// <para>
+		/// If set to <see cref="CascadeMode.Continue"/> then all rules in the class will execute regardless of failures.
+		/// </para>
+		/// <para>
+		/// If set to <see cref="CascadeMode.Stop"/> then execution of the validator will stop after any rule fails.
+		/// </para>
+		/// <para>
+		/// Note that cascade behaviour <i>within</i> individual rules is controlled by <see cref="RuleLevelCascadeMode"/>.
+		/// </para>
+		/// <para>
+		/// Do not set this to <see cref="CascadeMode.StopOnFirstFailure"/>.
+		/// <see cref="CascadeMode.StopOnFirstFailure"/> will be removed in a future release
+		/// and will throw an exception if used here.
+		/// </para>
+		/// </summary>
+		public CascadeMode ClassLevelCascadeMode {
+			get => _classLevelCascadeMode();
+			set {
+#pragma warning disable 618
+				if (value == CascadeMode.StopOnFirstFailure) {
+#pragma warning restore 618
+					throw new ArgumentException(_stopOnFirstFailureArgumentMessage);

```

Comment:
I think I'd prefer if this did a silent conversion to `CascadeMode.Stop` rather than throwing an exception

Reply:
Sure, I've changed all of these.